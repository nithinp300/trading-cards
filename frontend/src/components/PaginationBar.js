import React from 'react';
import Pagination from 'react-bootstrap/Pagination'

export default function PaginationBar(props){
  return(
    <div style={{margin:"auto"}}>
      <Pagination>
        <Pagination.First disabled={props.current_page === 1} 
        onClick={() => props.handlePaginationClick(1)}/>
        <Pagination.Prev disabled={props.current_page === 1}
        onClick={() => props.handlePaginationClick(props.current_page-1)}/>
        <Pagination.Item>{props.current_page}</Pagination.Item>
        <Pagination.Next disabled={props.current_page === props.last_page} 
        onClick={() => props.handlePaginationClick(props.current_page+1)}/>
        <Pagination.Last disabled={props.current_page === props.last_page} 
        onClick={() => props.handlePaginationClick(props.last_page)}/>
      </Pagination>
    </div>
  );
}