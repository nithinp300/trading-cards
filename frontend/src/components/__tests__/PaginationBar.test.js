import React from 'react';
import { shallow } from 'enzyme';
import PaginationBar from '../PaginationBar';

describe('PaginationBar', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PaginationBar />);
    expect(wrapper).toMatchSnapshot();
    // On the first run of this test, Jest will 
    // generate a snapshot file automatically.
  });
});