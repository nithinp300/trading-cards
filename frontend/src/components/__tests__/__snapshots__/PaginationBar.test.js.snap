// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationBar renders correctly 1`] = `ShallowWrapper {}`;
